### ISSUE TITLE
Handle nested JSON Arrays in deserialization

### ISSUE THREAD

**Issue by gdfast** [jgaa/restc-cpp#64]
Deserializing nested JSON arrays seems to not be supported right now, although nested arrays are valid JSON.

An example would be the results of an influxdb query (particularly the `values` object):

```json
{"results":[{"statement_id":0,"series":[{"name":"mymeas","columns":["time","myfield","mytag1","mytag2"],"values":[["2017-03-01T00:16:18Z",33.1,null,null],["2017-03-01T00:17:18Z",12.4,"12","14"]]}]}]}
```

I added a unit test in restc-cpp/tests/unit/JsonSerializeTests.cpp to demonstrate the issue

```cpp
STARTCASE(DeserializeNestedArray) {
    std::string json = R"([[1, 2, 3],[4, 5, 6]])";

    std::vector<std::vector<int>> nested_ints;
    RapidJsonDeserializer<decltype(nested_ints)> handler(nested_ints);
    Reader reader;
    StringStream ss(json.c_str());
    reader.Parse(ss, handler);

    CHECK_EQUAL(2, static_cast<int>(nested_ints.size()));

    for(int i = 0; i < 2; ++i) {
        CHECK_EQUAL(3, static_cast<int>(nested_ints[i].size()));

        for(int j = 0; j < 3; ++j) {
            CHECK_EQUAL(i * 3 + j + 1, nested_ints[i][j]);
        }
    }
} ENDCASE
```

```
[2018-05-18 14:38:51.296883] [0x000007f4d80da188] [debug]   ================================
[2018-05-18 14:38:51.296891] [0x000007f4d80da188] [info]    Test case: DeserializeNestedArray
[2018-05-18 14:38:51.296897] [0x000007f4d80da188] [debug]   ================================
json_serialize_tests: restc-cpp/include/restc-cpp/SerializeJson.h:1079: bool restc_cpp::RapidJsonDeserializer<T>::DoStartArray() [with T = std::vector<std::vector<int> >]: Assertion `false' failed.
Aborted (core dumped)
```

**Comment by gdfast** [comment]
By the way, I can confirm _serializing_ works fine... I added the ran the following test and it passed:
```cpp
STARTCASE(SerializeNestedVector)
{
    std::vector<std::vector<int>> nested_ints = {{-1,2,3},{4,5,-6}};

    StringBuffer s;
    Writer<StringBuffer> writer(s);

    RapidJsonSerializer<decltype(nested_ints), decltype(writer)>
        serializer(nested_ints, writer);

    serializer.Serialize();

    CHECK_EQUAL(R"([[-1,2,3],[4,5,-6]])"s,
                s.GetString());

} ENDCASE
```

**Comment by jgaa** [comment]
This was fixed in commit 6a47d0c8ee39f89108869b247c65ee594296fc4
